packages/simulation-sdk/src/operations.ts

```diff
@@ -301,11 +306,21 @@ export interface Erc20OperationArgs {
     spender: Address;
     amount: bigint;
     nonce: bigint;
+    /**
+     * The maximum block timestamp (included) for which the permit will be signed and valid.
+     * Defaults to the simulation state's timestamp + 2h.
+     */
+    deadline?: bigint;
```

Comment:
wouldn't it be more convenient to only ask for the validity of the signature (default to 2H) so that we don't need to check

Reply:
this could lead to an overflow though if we don't take care

Reply:
But as a consumer i do want to set an absolute deadline so my simulation reverts for example if my bundle is too old (example: i leave my interface open for 2h and state did not update for some reason)

Reply:
My suggestion was just to let the user define a duration instead of the deadline:
so that the deadline would be computed in the sdk as : `deadline = Time.timestamp() + duration` instead of directly specifying the `deadline`. This would save us from having to check that `deadline >= Time.timestamp()` everywhere.

Reply:
This is not that important